app.js

```diff
@@ -0,0 +1,174 @@
+var notesList = document.querySelector(".notes-list");
+var noteTitle = document.querySelector(".note-title");
+var notenote = document.querySelector(".note-note");
+var noteItem = document.querySelector(".note-item");
+var notenote = document.querySelector(".note-note");
+var emptyState = document.querySelector(".empty-state");
+
+var selectedNote = 0;
+var notes = [];
+
+toggleSwitch = document.querySelector('.theme-switch input[type="checkbox"]');
+
+function switchTheme(e) {
+  if (e.target.checked) {
+    document.documentElement.setAttribute("data-theme", "dark");
+    localStorage.setItem("theme", "dark");
+  } else {
+    document.documentElement.setAttribute("data-theme", "light");
+    localStorage.setItem("theme", "light");
+  }
+}
+
+if (currentTheme) {
+  toggleSwitch.checked = true;
+}
+
+toggleSwitch.addEventListener("change", switchTheme, false);
+
+function initializeNotes() {
+  var notesArr = localStorage.getItem(["notes"]);
+  if (JSON.parse(notesArr) == null || JSON.parse(notesArr).length === 0) {
```

Comment:
instead of parsing twice, you can parse once and store it in a variable. Also, on first start `notesArr` would be undefined, and hence JSON.parse will fail with error, won't it?

Reply:
Yup, using variable would be better.

And `localStorage.getItem()` returns null if the data we want isn't store in local storage. And hence, on first start, `notesArr` will be null and `JSON.parse()` will return null as well.